### PR TITLE
Add benchmark/ips for perf comparison 

### PR DESCRIPTION
**Pull request by Funfun**
i found quite representative to use https://github.com/evanphx/benchmark-ips for some benchmark tests.

### CODE REVIEW

**Comment by johnnyshields** [comment]
@Funfun can you rebase this PR please?

**Comment by estolfo** [comment]
@Funfun Can you rebase please?

**Comment by Funfun** [comment]
@estolfo yeah, sure

**Comment by estolfo** [comment]
thanks!

File: perf/benchmark_ips.rb [hunk 1]
```diff
@@ -0,0 +1,273 @@
+require "benchmark/ips"
+require "mongoid"
```
Comment (estolfo) [inline perf/benchmark_ips.rb]: we should require mongoid from this repo so the repo's code is tested.

File: perf/benchmark_ips.rb [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+require "./perf/models"
+require './perf/gc_suite'
+
+Mongoid.connect_to("mongoid_perf_test")
+
```
Comment (estolfo) [inline perf/benchmark_ips.rb]: Put the logging level to WARN so time isn't wasted printing log lines:

```ruby
Mongo::Logger.logger.level = ::Logger::WARN
```

File: perf/benchmark_ips_iteration_cases.rb [hunk 3]
```diff
@@ -0,0 +1,76 @@
+require "benchmark/ips"
+require "mongoid"
+require "./perf/models"
+require './perf/gc_suite'
+
+Mongoid.connect_to("mongoid_perf_test")
+Mongoid.purge!
+
```
Comment (estolfo) [inline perf/benchmark_ips_iteration_cases.rb]: Put the logging level to WARN so time isn't wasted printing log lines:

```ruby
Mongo::Logger.logger.level = ::Logger::WARN
```

File: perf/benchmark_ips_iteration_cases.rb [hunk 4]
```diff
@@ -0,0 +1,76 @@
+require "benchmark/ips"
+require "mongoid"
```
Comment (estolfo) [inline perf/benchmark_ips_iteration_cases.rb]: same here: we should do a relative require so the repo's code is tested.

**Comment by johnnyshields** [comment]
@estolfo can you take ownership of this PR, fix it an merge it? Original author raised 2 years ago and has probably abandoned it.

**Pull request merged**